services/search/generate_docs_embeddings.py

```diff
@@ -0,0 +1,144 @@
+import os
+import time
+import glob
+import argparse
+from dotenv import load_dotenv
+from langchain_openai import OpenAIEmbeddings
+from langchain_experimental.text_splitter import SemanticChunker
+from pymilvus import FieldSchema, CollectionSchema, DataType, utility, Collection, connections
+
+load_dotenv()
+
+def read_md_files(directories):
+    docs = []
+    for directory in directories:
+        md_files = glob.glob(f"{directory}/**/*.md", recursive=True)
+        for file in md_files:
+            with open(file, "r", encoding="utf-8") as f:
+                docs.append((file, f.read()))  # Returning a tuple with file name and content
+    return docs
+
+def split_md_by_sections(file_name, content, embed_model):
+    # Initialize SemanticChunker directly
+    semantic_splitter = SemanticChunker(
+        embeddings=embed_model,
+        breakpoint_threshold_type="gradient",
+        breakpoint_threshold_amount=80
+    )
+    
+    # Convert content to a single document format expected by SemanticChunker
+    document = {"page_content": content}
+    splits = semantic_splitter.create_documents([document["page_content"]])
+
+    # Write the sections to disk
+    output_dir = "./tmp/split_sections"
+    os.makedirs(output_dir, exist_ok=True)
+    with open(os.path.join(output_dir, f"{os.path.basename(file_name)}_sections.md"), "w", encoding="utf-8") as out_file:
+        for section in splits:
+            out_file.write(f"## Section Start: " + "\n------\n")
+            out_file.write(section.page_content + "\n------\n")
+
+    # Calculate the max split length
+    max_split_length = max(len(section.page_content) for section in splits)
+
+    return splits, max_split_length
+
+if __name__ == "__main__":
+    # Parse the command line arguments - repo_path and collection_name
+    parser = argparse.ArgumentParser(description="Initialize Milvus with markdown files")  
+    parser.add_argument("repo_path", type=str, help="Path to the repository containing markdown files")
+    parser.add_argument("collection_name", type=str, help="Name of the Milvus collection")
+    args = parser.parse_args()
+
+    openai_key = os.getenv("OPENAI_API_KEY")
+    openai_ef = OpenAIEmbeddings(
+        model="text-embedding-3-small", 
+        api_key=openai_key
+    )
+
+    # Combine the base path with the additional folders
+    docs_to_embed = ["docs/jobs"]
```

Comment:
This list is hard-coded, so if we ever want to update the list of paths, we have to call a python expert in to find the right line of code.

This should really be in a config file, like I asked, with one path per line. Bonus points if this config file accepts standard glob patterns.

The config file needs to be fully documented in the readme so that  developers unfamiliar with the codebase can update the paths a year from now.